ThunderCloud/ContentController.swift

```diff
@@ -366,37 +366,39 @@ public class ContentController: NSObject {
     
     /// Downloads a full storm content bundle, this will clear all directories and will also mark the downloaded bundle as the 'initial' bundle timestamp
     /// so that we can avoid downloading any post-landmark publishes from content-available notifications!
-    /// - Parameter progressHandler: A closure which will be called with the progress of the bundle download
-    public func downloadFullBundle(with progressHandler: ContentUpdateProgressHandler?) {
+    /// - Parameter buildTimestamp: The timestamp of the build, used to make sure we don't bypass any landmark publishes
+    /// - Parameter progressHandler: A closure called when as the download progresses
+    public func downloadFullBundle(buildTimestamp: TimeInterval? = nil, with progressHandler: ContentUpdateProgressHandler?) {
         
-        //Clear existing bundles first
-        if let _currentBundle = bundleDirectory {
-            removeBundle(in: _currentBundle)
-        }
+        removeAllContentBundles()
+        configureBaseURL()
         
-        if let _deltaBundle = deltaDirectory {
-            removeBundle(in: _deltaBundle)
-        }
+        let stormAppId = UserDefaults.standard.string(forKey: "TSCAppId") ?? Storm.API.AppID
         
-        if let _tempDirectory = temporaryUpdateDirectory {
-            removeBundle(in: _tempDirectory)
+        guard let baseString = Storm.API.BaseURL, let version = Storm.API.Version, let appId = stormAppId else {
+            fatalError("Failed to get required parameters to download a storm bundle: missing one of Base URL, Api Version or AppID")
         }
         
-        // Remove intial bundle timestamp as we've now cleared out all evidence of any bundles!
-        initialBundleTimestamp = nil
-        
-        configureBaseURL()
+        guard var urlComponents = URLComponents(string: "\(baseString)/\(version)/apps/\(appId)/bundle") else {
+            fatalError("Failed to create url to download a storm bundle: one of Base URL, Api Version or AppID are invalid url parts")
+        }
+            
+        guard let bundleDirectory = bundleDirectory else {
+            return
```

Comment:
What does it mean if we get here?

Reply:
Yeah I was unsure here whether to `fatalError` or not... it basically means the directory which stores the default content failed to be created:

```
//Identify folders for bundle
            if let _bundlePath = NSSearchPathForDirectoriesInDomains(.applicationSupportDirectory, .userDomainMask, true).last {
                
                let _bundleDirectory = URL(fileURLWithPath: _bundlePath, isDirectory: true).appendingPathComponent("StormBundle")
                
                bundleDirectory = _bundleDirectory
                
                //Create application support directory
                do {
                    try FileManager.default.createDirectory(atPath: _bundleDirectory.path, withIntermediateDirectories: true, attributes: nil)
                } catch {
                    baymax_log("Failed to create bundle directory at \(_bundleDirectory.absoluteString)", subsystem: Logger.stormSubsystem, category: ContentController.logCategory, type: .fault)
                    os_log("Failed to create bundle directory at %@", log: contentControllerLog, type: .fault, _bundleDirectory.absoluteString)
                }
            }
```

I didn't feel like here was a valid place to `fatalError` as everywhere else that uses this property just fails silently

Reply:
I guess in this case will could try and re-download the bundle? Possibly by going down the root of first app launch